noxfile.py

```diff
@@ -113,3 +114,28 @@ def clone_core(session: nox.Session):
     source tree to facilitate building docs.
     """
     session.run_always("python", "docs/bin/clone-core.py")
+
+
+checker_tests = [
+    path.with_suffix("").name for path in Path("tests/checkers/").glob("*.py")
+]
+
+
+@nox.session
+@nox.parametrize(["test"], checker_tests, checker_tests)
+def checkers(session: nox.Session, test: str):
+    """
+    Run docs build checkers
+    """
+    parser = ArgumentParser(prog=f"nox -e {session.name} --")
+    parser.add_argument(
+        "--relaxed",
+        default=False,
+        action=BooleanOptionalAction,
+        help="Whether to use requirements-relaxed file. (Default: %(default)s)",
+    )
+    args = parser.parse_args(session.posargs)
+
+    install(session, req="requirements-relaxed" if args.relaxed else "requirements")
```

Comment:
Could you please remind me why we don't include `-c requirements.txt` here? Including the pinned versions with the test would make sense to me and be consistent with the "docs.ansible.com" build.

Reply:
Our custom `install()` function already handles all that.

https://github.com/ansible/ansible-documentation/pull/668/files/1ab7527f41bc14720928b72523c0b211aac7b88b#diff-f7a16a65f061822bcc73b8296f4dc837353d379d8d9cc5307982cb6941442835R21-R22

Reply:
Ah, thanks. TBH this seems a little tricky and would be nice if we could make it a little more obvious for users as Sviat mentioned in his initial review here: https://github.com/ansible/ansible-documentation/pull/258/files#r1288678876

I've been playing around with this some but thought I'd just ask. I set `PINNED=True` and expect to see `-c tests/requirements.txt` arguments when `PIP_CONSTRAINT` is set.

```
if PINNED:
        pip_constraint = f"tests/{req}.txt"
        print(f"Setting PIP_CONSTRAINT to {pip_constraint}")
        kwargs.setdefault("env", {})["PIP_CONSTRAINT"] = pip_constraint
```

Now when I run the nox session I can see:

```
Setting PIP_CONSTRAINT to tests/requirements.txt
nox > python -m pip install -r tests/requirements.in
```

Does this make sense? I'm not suggesting we modify the `install()` function by the way. I just think we should make it really clear how to test against dependencies as in https://docs.ansible.com/ansible/latest/community/documentation_contributions.html#setting-up-your-environment-to-build-documentation-locally

Agree this isn't something that you should worry about too much on the devel branch. But this feels like something we should address while we're in the docs.

Reply:
Pinned dependencies are used by default. Let's create a separate PR (to avoid scope creep on this one) to add `session.log(f"export PIP_CONSTRAINT={pip_constraint}")` to the `install()` function. We can also document the PINNED environment variable. Does that sound good?

Reply:
ah... `session.log` is the better way. yeah, that sounds good to me. thanks for putting up with my questions here. I know they might be a little obvious to folks with a lot more nox experience.

Reply:
No, not really :). Your questions are much appreciated!